src/handlers/records-write.ts

```diff
@@ -79,23 +81,34 @@ export class RecordsWriteHandler implements MethodHandler {
 
     const isLatestBaseState = true;
     const indexes = await constructRecordsWriteIndexes(recordsWrite, isLatestBaseState);
-
-    try {
-      // try to store data, unless options explicitly say to skip storage
-      if (options === undefined || !options.skipDataStorage) {
-        await this.putData(tenant, message, dataStream, newestExistingMessage as (RecordsWriteMessage|RecordsDeleteMessage) | undefined);
-      }
-    } catch (error) {
-      const e = error as any;
-      if (e.code === DwnErrorCode.RecordsWriteMissingDataStream ||
-          e.code === DwnErrorCode.RecordsWriteMissingData ||
-          e.code === DwnErrorCode.RecordsWriteDataCidMismatch ||
-          e.code === DwnErrorCode.RecordsWriteDataSizeMismatch) {
-        return messageReplyFromError(error, 400);
+    // try to store data, unless options explicitly say to skip storage
+    if (options === undefined || !options.skipDataStorage) {
+      if (dataStream === undefined && newestExistingMessage?.descriptor.method === DwnMethodName.Delete) {
+        return messageReplyFromError(new DwnError(DwnErrorCode.RecordsWriteMissingDataStream, 'No data stream was provided with the previous message being a delete'), 400);
       }
 
-      // else throw
-      throw error;
+      try {
+        if (message.descriptor.dataSize <= DwnConstant.maxDataSizeAllowedToBeEncoded) {
+          message = await this.processEncodedData(
```

Comment:
Can we use a variable other than `message`? I try to stay away from modifying function parameters, especially for a long one like this.
```suggestion
          messageWithOptionalEncodedData = await this.processEncodedData(
```